proto/src/determined/api/v1/api.proto

```diff
@@ -509,6 +509,78 @@ service Determined {
       tags: "Internal"
     };
   }
+  // Get the absolute training length until the searcher needs the trial to
+  // report a validation.
+  rpc GetTrialSearcherTrainUntil(GetTrialSearcherTrainUntilRequest)
+      returns (GetTrialSearcherTrainUntilResponse) {
+    option (google.api.http) = {
+      get: "/api/v1/trials/{trial_id}/searcher/train_until"
```

Comment:
A trial shouldn't be aware of the existence of the searcher, right? It should just know it needs to early stop.
```suggestion
      get: "/api/v1/trials/{trial_id}/train_length"
```

Reply:
@rb-determined-ai and I talked about this, he felt the same so yeah, this is fine with me.